### PR TITLE
 Handle property groups round trip vCard <-> jCard according to spec

### PR DESCRIPTION
**Pull request by darktrojan**
Extends #442 with nits fixed and a test that ical is handled differently.

### CODE REVIEW

File: test/parser/grouped.ics [hunk 1]
```diff
@@ -0,0 +1,3 @@
+BEGIN:VCALENDAR
+GROUP1.X-TEST;VALUE=TEXT:test!
```
Comment (darktrojan) [inline test/parser/grouped.ics:2]: This isn't valid iCal but it does prove the handling difference between iCal and vCard.

File: lib/ical/parse.js [hunk 2]
```diff
@@ -232,9 +232,20 @@ ICAL.parse = (function() {
     var multiValue = false;
     var structuredValue = false;
     var propertyDetails;
+    var splitName;
+    var ungroupedName;
+
+    // fetch the ungrouped part of the name
+    if (name.indexOf('.') !== -1 && state.designSet.propertyGroups) {
```
Comment (kewisch) [inline lib/ical/parse.js]: Minor nit, let's turn this around. Slight perf increase in skipping the indexOf for iCalendar:
```suggestion
    if (state.designSet.propertyGroups && name.indexOf('.') !== -1) {
```

File: lib/ical/stringify.js [hunk 3]
```diff
@@ -108,10 +108,24 @@ ICAL.stringify = (function() {
...
+    var line;
+    if (groupName && designSet.propertyGroups) {
+      line = groupName.toUpperCase() + "." + name;
+    } else {
+      line = name;
+    }
 
     var paramName;
     for (paramName in params) {
+      if (paramName == 'group' && designSet.propertyGroups) {
```
Comment (kewisch) [inline lib/ical/stringify.js]:
```suggestion
      if (designSet.propertyGroups && paramName == 'group') {
```

**Review by kewisch — APPROVED** [review]
r+wc, feel free to squash and merge.

**Comment by darktrojan** [comment]
> r+wc, feel free to squash and merge.

I don't have write access. I've also left the original two commits unsquashed because they're not mine and it doesn't feel right to hide somebody else's work.

**Pull request merged**